fhir-path/src/main/java/com/ibm/fhir/path/util/FHIRPathUtil.java

```diff
@@ -270,41 +302,166 @@ public static boolean hasQuantityValue(Collection<FHIRPathNode> nodes) {
     }
 
     public static boolean evaluatesToBoolean(Collection<FHIRPathNode> nodes) {
-        return isTrue(nodes) || isFalse(nodes);
+        return evaluatesToTrue(nodes) || isFalse(nodes);
+    }
+
+    /**
+     * Indicates whether the collection of nodes evaluates to a boolean per:
+     * <a href="http://hl7.org/fhirpath/N1/index.html#singleton-evaluation-of-collections">Singleton Evaluation of Collections</a>
+     *
+     * @param nodes
+     *    the collection of nodes
+     * @return
+     *    true if the collection of nodes evaluates to a boolean, false otherwise
+     */
+    public static boolean evaluatesToTrue(Collection<FHIRPathNode> nodes) {
+        if (convertsToBoolean(nodes)) {
+            return toBoolean(nodes) == true;
+        }
+        return isSingleton(nodes);
     }
 
     public static boolean isTrue(Collection<FHIRPathNode> nodes) {
```

Comment:
could use some javadoc to further distinguish it from evaluatesToTrue;  this one would basically just say it returns true iff it convertsToBoolean and has a true value

Reply:
Added Javadoc